apollo-portal/src/main/java/com/ctrip/framework/apollo/portal/service/ConfigsImportService.java

```diff
@@ -16,122 +16,460 @@
   }
 
-  /**
-   * @see ConfigsImportService#importOneConfigFromText(java.lang.String, java.lang.String, java.lang.String)
-   * @throws AccessControlException if has no modify namespace permission
-   */
-  public void importOneConfigFromFile(
-      final String env,
-      final String standardFilename,
-      final InputStream inputStream
-  ) {
-    final String configText;
-    try(InputStream in = inputStream) {
-      configText = ConfigToFileUtils.fileToString(in);
+  private void importNamespace(final String appId, final Env env,
+                               final String clusterName, final String namespaceName,
+                               final String configText, final String format,
+                               boolean ignoreConflictNamespace) {
+    NamespaceDTO namespaceDTO;
+    try {
+      namespaceDTO = namespaceService.loadNamespaceBaseInfo(appId, env, clusterName, namespaceName);
+    } catch (Exception e) {
+      //not existed
+      namespaceDTO = null;
+    }
+
+    if (namespaceDTO == null) {
+      namespaceDTO = new NamespaceDTO();
+      namespaceDTO.setAppId(appId);
+      namespaceDTO.setClusterName(clusterName);
+      namespaceDTO.setNamespaceName(namespaceName);
+      namespaceDTO.setDataChangeCreatedBy(currentUser);
+      namespaceDTO.setDataChangeLastModifiedBy(currentUser);
+      namespaceDTO = namespaceService.createNamespace(env, namespaceDTO);
+
+      roleInitializationService.initNamespaceRoles(appId, namespaceName, currentUser);
+      roleInitializationService.initNamespaceEnvRoles(appId, namespaceName, currentUser);
+    }
+
+    List<ItemDTO> itemDTOS = itemService.findItems(appId, env, clusterName, namespaceName);
+    // skip import if target namespace has existed items
+    if (!CollectionUtils.isEmpty(itemDTOS) && ignoreConflictNamespace) {
+      return;
+    }
+
+    importItems(appId, env, clusterName, namespaceName, configText, namespaceDTO);
+  }
+
+  private void importItems(String appId, Env env, String clusterName, String namespaceName, String configText,
+                           NamespaceDTO namespaceDTO) {
+    List<ItemDTO> toImportItems = gson.fromJson(configText, GsonType.ITEM_DTOS);
+
+    toImportItems.parallelStream().forEach(newItem -> {
+      String key = newItem.getKey();
+      newItem.setNamespaceId(namespaceDTO.getId());
+      newItem.setDataChangeCreatedBy(currentUser);
+      newItem.setDataChangeLastModifiedBy(currentUser);
+      newItem.setDataChangeCreatedTime(new Date());
+      newItem.setDataChangeLastModifiedTime(new Date());
+
+      try {
+        ItemDTO oldItem = itemService.loadItem(env, appId, clusterName, namespaceName, key);
+        newItem.setId(oldItem.getId());
+        //existed
+        itemService.updateItem(appId, env, clusterName, namespaceName, newItem);
+      } catch (Exception e) {
+        //not existed
+        itemService.createItem(appId, env, clusterName, namespaceName, newItem);
```

Comment:
There could be many exceptions and shall we only fallback to create item when the item is not existed?
```suggestion
    if (e instanceof HttpStatusCodeException && ((HttpStatusCodeException) e).getStatusCode().equals(HttpStatus.NOT_FOUND)) {
        itemService.createItem(appId, env, clusterName, namespaceName, newItem);
    }
```

Reply:
BTW, if there are more than 2 comment lines, then only the first comment will be imported, e.g.

```properties
k1 = v1
# comment1
k2 = v2
# comment2
```

Reply:
Support create text comment.